drivers/docker/driver_unix_test.go

```diff
@@ -1,98 +1,735 @@
-// +build !windows
+// +build darwin dragonfly freebsd linux netbsd openbsd solaris
 
 package docker
 
 import (
-	"context"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"os"
 	"path/filepath"
+	"sort"
+	"strconv"
 	"strings"
 	"testing"
 	"time"
 
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/hashicorp/nomad/client/allocdir"
 	"github.com/hashicorp/nomad/client/testutil"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/plugins/drivers"
 	tu "github.com/hashicorp/nomad/testutil"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func TestDockerDriver_Signal(t *testing.T) {
+func TestDockerDriver_User(t *testing.T) {
 	if !tu.IsTravis() {
 		t.Parallel()
 	}
-	if !testutil.DockerIsConnected(t) {
-		t.Skip("Docker not connected")
-	}
-
+	testutil.DockerCompatible(t)
 	task, cfg, _ := dockerTask(t)
-	cfg.Command = "/bin/sh"
-	cfg.Args = []string{"local/test.sh"}
+	task.User = "alice"
+	cfg.Command = "/bin/sleep"
+	cfg.Args = []string{"10000"}
 	require.NoError(t, task.EncodeConcreteDriverConfig(cfg))
 
-	driver := dockerDriverHarness(t, nil)
-	cleanup := driver.MkAllocDir(task, true)
+	d := dockerDriverHarness(t, nil)
+	cleanup := d.MkAllocDir(task, true)
 	defer cleanup()
+	copyImage(t, task.TaskDir(), "busybox.tar")
+
+	_, _, err := d.StartTask(task)
+	if err == nil {
+		d.DestroyTask(task.ID, true)
```

Comment:
Do we need to destroy the task even if it fails?  Also, `require.Error(...)` ?

Reply:
Blindly copy/pasted this test and I'm trying to avoid changing more behaviour rn